### DISCUSSION TITLE
How to detect mouse eye and quantify data

### DISCUSSION THREAD

**Discussion by kairos-bar** [bonsai-rx/bonsai#1435] [Q&A]
Hello! I'm trying to detect mouse eye when they blink their eyes. So I use bonsai program to detect that.
First, I enlarged the mouse's eyes in the video. 
After that, I want to quantify the change in the size of my eyes when I give a strong wind.
The problem here is that standardization cannot be performed because each mouse eye has a different size and location. And I don't know how to measure the size change exactly.
![eye capture](https://github.com/bonsai-rx/bonsai/assets/137870547/01d9978c-475c-4032-9996-6502af374e26)
This is actually what I've done so far.
Please give us a lot of help... Thank you......

**Comment by bruno-f-cruz** [comment]
Heya,
Can you include your workflow and some sample frames? Its hard to help without a bit of context. If you want to generally detect the eye here are some possibilities:

- Use background subtraction. If you have an image of your background, even if each mouse is in a different location, you can first detect where the background changes and use that to "seed" the rest of the algorithm
- Use a machine-learning approach (DLC or SLEAP) to detect the eye for a few frames at the start of your workflow, and then switch to your computer vision routine.
- Use "shape detectors" available in bonsai to detect round objects (e.g. Hough detector)

Hope this helps,
Bruno
